OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -89,6 +89,32 @@ static class DreamProcNativeRoot {
             return new DreamValue(key);
         }
 
+        [DreamProc("clamp")]
+        [DreamProcParameter("Value", Type = DreamValueType.Number | DreamValueType.DreamObject)]
+        [DreamProcParameter("Low", Type = DreamValueType.Number)]
+        [DreamProcParameter("High", Type = DreamValueType.Number)]
+        public static DreamValue NativeProc_clamp(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            DreamValue value = arguments.GetArgument(0, "Value");
+            float lVal = arguments.GetArgument(1, "Low").GetValueAsNumber();
+            float hVal = arguments.GetArgument(2, "High").GetValueAsNumber();
+
+            if (value.TryGetValueAsDreamList(out DreamList list))
+            {
+                DreamList tmp = new DreamList();
+                for (int i = 1; i < list.GetLength(); i++)
+                {
+                    DreamValue val = list.GetValues()[i];
+                    tmp.AddValue(new DreamValue((float)Math.Clamp(val.GetValueAsNumber(), lVal, hVal)));
+                }
+                return new DreamValue(tmp);
+            }
+            else
+            {
+                return new DreamValue((float)Math.Clamp(value.GetValueAsNumber(), lVal, hVal));
```

Comment:
`Math.Clamp` already returns a float if all three arguments are floats.
```suggestion
                return new DreamValue(Math.Clamp(value.GetValueAsNumber(), lVal, hVal));
```